src/contracts/deposit/deposit-tree/deposit-tree.ts

```diff
@@ -0,0 +1,138 @@
+import { ethers } from 'ethers';
+import { digest2Bytes32 } from '@chainsafe/as-sha256';
+import { fromHexString } from '@chainsafe/ssz';
+import { parseLittleEndian64, toLittleEndian64 } from '../deposit.utils';
+import { DepositData } from 'bls/bls.containers';
+import { NodeData } from '../interfaces';
+
+export class DepositTree {
+  static DEPOSIT_CONTRACT_TREE_DEPTH = 32;
+  static ZERO_HASH = fromHexString(
+    '0x0000000000000000000000000000000000000000000000000000000000000000',
+  );
+  zeroHashes: Uint8Array[] = new Array(DepositTree.DEPOSIT_CONTRACT_TREE_DEPTH);
+  branch: Uint8Array[] = [];
+  nodeCount = 0;
+
+  constructor() {
+    this.formZeroHashes();
+  }
+
+  /**
+   * Initializes the zero hashes used in the tree.
+   */
+  private formZeroHashes() {
+    this.zeroHashes[0] = DepositTree.ZERO_HASH;
+    for (
+      let height = 0;
+      height < DepositTree.DEPOSIT_CONTRACT_TREE_DEPTH - 1;
+      height++
+    ) {
+      this.zeroHashes[height + 1] = digest2Bytes32(
+        this.zeroHashes[height],
+        this.zeroHashes[height],
+      );
+    }
+  }
+
+  /**
+   * Forms the branch of the tree needed to update the root when a new node is inserted.
+   * @param {Uint8Array} node - The node's data to be inserted.
+   * @param {number} depositCount - The sequential index of the deposit, representing the total deposits.
+   * @returns {Uint8Array[] | undefined} The updated branch of the tree after inserting the node.
+   */
+  private formBranch(
+    node: Uint8Array,
+    depositCount: number,
+  ): Uint8Array[] | undefined {
+    let size = depositCount;
+    for (
+      let height = 0;
+      height < DepositTree.DEPOSIT_CONTRACT_TREE_DEPTH;
+      height++
+    ) {
+      if ((size & 1) == 1) {
+        this.branch[height] = node;
+        return this.branch;
+      }
+
+      node = digest2Bytes32(this.branch[height], node);
```

Comment:
look like this.branch[height] can be undefined

Reply:
maybe here should be smth like 

if (this.branch[height] === undefined) {
      this.branch[height] = this.zeroHashes[height];
 }